src/main/kotlin/io/snyk/plugin/snykcode/SnykCodeBulkFileListener.kt

```diff
@@ -55,13 +68,44 @@ class SnykCodeBulkFileListener : SnykBulkFileListener() {
     }
 
     override fun after(project: Project, virtualFilesAffected: Set<VirtualFile>) {
-        // update .dcignore caches if needed
-        SnykCodeIgnoreInfoHolder.instance.updateIgnoreFileCachesIfAffected(
-            toSnykCodeFileSet(
-                project,
-                virtualFilesAffected
-            )
-        )
+        val filesAffected = toSnykCodeFileSet(project, virtualFilesAffected)
+
+        if (isSnykCodeLSEnabled()) {
+            processAfterForLS(filesAffected, project)
+            return
+        }
+        /* update .dcignore caches if needed */
+        SnykCodeIgnoreInfoHolder.instance.updateIgnoreFileCachesIfAffected(filesAffected)
+    }
+
+    override fun forwardEvents(events: MutableList<out VFileEvent>) {
+        if (!isSnykCodeLSEnabled()) return
+        LanguageServerWrapper.getInstance().ensureLanguageServerInitialized()
+        val languageServer = LanguageServerWrapper.getInstance().languageServer
+        for (event in events) {
+            if (event.file == null || !event.isFromSave) continue
+            val file = event.file!!
```

Comment:
`event.file == null` check doesn't make the next `event.file!!` redundant? 
Excuse my little Kotlin expertise 🙏 

https://kotlinlang.org/docs/null-safety.html#the-operator

Reply:
It doesn't, the compiler somehow does not catch the null check, most likely because of the second condition.

Reply:
Yeah, I was sure, the compiler complained before, but removing it seems to work now :).